File: Cpp/41_First_Missing_Positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int n = nums.size();
        vector<bool> visit(n+1, false);
        for(auto& num : nums){
            if(num < 0 || num > n) continue;
            visit[num] = true;
        }
        
        for(int i=1; i<visit.size(); ++i){
            if(!visit[i]) return i;
        }
        
        return n+1;
    }
};

File: Kotlin/41_First_Missing_Positive.kt
class Solution {
    fun firstMissingPositive(nums: IntArray): Int {
        val s = nums.filter { it > 0 }.toSet()
        
        for(i in 1..nums.size) {
            if (!s.contains(i)) return i
        }
        
        return nums.size + 1
    }
}